azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -748,7 +766,9 @@ func resourceArmKubernetesClusterUpdate(d *schema.ResourceData, meta interface{}
 	if d.HasChange("api_server_authorized_ip_ranges") {
 		updateCluster = true
 		apiServerAuthorizedIPRangesRaw := d.Get("api_server_authorized_ip_ranges").(*schema.Set).List()
-		existing.APIServerAuthorizedIPRanges = utils.ExpandStringSlice(apiServerAuthorizedIPRangesRaw)
+		existing.ManagedClusterProperties.APIServerAccessProfile = &containerservice.ManagedClusterAPIServerAccessProfile{
+			AuthorizedIPRanges: utils.ExpandStringSlice(apiServerAuthorizedIPRangesRaw),
```

Comment:
presumably we also need to pass in `private_link_enabled` here too, even if it's not updatable, so when updating a cluster with a value of `true` the API doesn't return an error?

Reply:
I've added it just in case. Also added to the acc tests (Ran OK):
1) Create and Update with privateLink On
2) Create and Update with privateLink Off.